fl4health/clients/tabular_data_client.py

```diff
@@ -0,0 +1,105 @@
+from fl4health.feature_alignment.tab_features_preprocessor import TabularFeaturesPreprocessor
+from fl4health.utils.metrics import Metric
+
+
+class TabularDataClient(BasicClient):
+    def __init__(
+        self, data_path: Path, metrics: Sequence[Metric], device: torch.device, id_column: str, target_column: str
+    ) -> None:
+        super().__init__(data_path, metrics, device)
+        self.tabular_features_info_encoder: TabularFeaturesInfoEncoder
+        self.tabular_features_preprocessor: TabularFeaturesPreprocessor
+        self.df: pd.DataFrame
+        self.input_dimension: int
+        self.output_dimension: int
+        self.id_column = id_column
+        self.target_column = target_column
+        # The aligned data and targets, which are used to construct dataloaders.
+        self.aligned_features: NDArray
+        self.aligned_targets: NDArray
+
+    def setup_client(self, config: Config) -> None:
+        """
+        Initialize the client by encoding the information of its tabular data
+        and initializing the corresponding TabularFeaturesPreprocessor.
+
+        config[FORMAT_SPECIFIED] indicates whether the server has obtained
+        the source of information to perform feature alignment.
+        If it is True, it means the server has obtained such information
+        (either a priori or by polling a client).
+        So the client will encode that information and use it instead
+        to perform feature preprocessing.
+        """
+        format_specified = self.narrow_config_type(config, FORMAT_SPECIFIED, bool)
+        self.df = self.get_data_frame(config)
+
+        if format_specified:
+            # Since the server has obtained its source of information,
+            # the client will encode that instead.
+            self.tabular_features_info_encoder = TabularFeaturesInfoEncoder.from_json(
+                self.narrow_config_type(config, FEATURE_INFO, str)
+            )
+            self.tabular_features_preprocessor = TabularFeaturesPreprocessor(self.tabular_features_info_encoder)
+
+            # preprocess features.
+            self.aligned_features, self.aligned_targets = self.tabular_features_preprocessor.preprocess_features(
+                self.df
+            )
+
+            # Obtain the input and output dimensions to be sent to
+            # the server for global model initialization.
+            self.input_dimension = self.aligned_features.shape[1]
+            self.output_dimension = self.tabular_features_info_encoder.get_target_dimension()
+            log(INFO, f"input dimension: {self.input_dimension}, output_dimension: {self.output_dimension}")
+
+            super().setup_client(config)
+
+            # freeing the memory of aligned features/targets.
+            del self.aligned_features
+            del self.aligned_targets
+        else:
+            # Encode the information of the client's local tabular data.
```

Comment:
Probably worth commenting that this is expected to happen only once and only if the client is requested to provide alignment information?